### ISSUE TITLE
Support nested modpacks in mainmenu

### ISSUE THREAD

**Issue by yyt16384** [minetest/minetest#3363]
Currently mainmenu treat second-level modpacks as mods, so their contents are not shown and you can't enable or disable them in the mainmenu. A `load_mod_<modpack_name> = true` line is added when they are enabled, which is just a no-op most of the time.

They work fine in subgames and also when they are enabled manually in `world.mt`, so clearly the core supports them. Therefore it makes sense to support them in mainmenu.

**Comment by C1ffisme** [comment]
I can see some use for this...

Although as a workaround you could just put all the mods in the 2nd modpack into the first one and append a string like ```mod1_``` to the front of each mod in the 2nd modpack.

**Comment by yyt16384** [comment]
@C1ffisme I don't think you can rename mod directories, or at least not until something like `mod.conf` gets standardized.

One possible use for this is to organize mods and modpacks in bigger modpacks, so they can be enabled/disabled together.

**Comment by ShadowNinja** [comment]
:+1: I never use the configure menu because of this issue (I have to symlinks in my mods directory that point to "modpacks" of my mods, some of which are themselves modpacks (Mesecons, Technic, WorldEdit, ...)
